File: benches/lang-comparison/go/fibonacci.go
// Fibonacci - recursive and iterative
package main

import "fmt"

func fibRec(n int64) int64 {
	if n <= 1 {
		return n
	}
	return fibRec(n-1) + fibRec(n-2)
}

func fibIter(n int64) int64 {
	var a, b int64 = 0, 1
	for i := int64(0); i < n; i++ {
		a, b = b, a+b
	}
	return a
}

func main() {
	fmt.Println(fibRec(20))
	fmt.Println(fibIter(50))
}

File: benches/lang-comparison/c/fibonacci.c
// Fibonacci - recursive and iterative
#include <stdio.h>
#include <stdint.h>

int64_t fib_rec(int64_t n) {
    if (n <= 1) return n;
    return fib_rec(n - 1) + fib_rec(n - 2);
}

int64_t fib_iter(int64_t n) {
    int64_t a = 0, b = 1;
    for (int64_t i = 0; i < n; i++) {
        int64_t t = a + b;
        a = b;
        b = t;
    }
    return a;
}

int main() {
    printf("%lld\n", fib_rec(20));
    printf("%lld\n", fib_iter(50));
    return 0;
}
